### PR TITLE
expression: NotNullFlag is not set correctly for some expression

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Problem Summary:
for some expression, the result type's NotNullFlag is not set correctly
### What is changed and how it works?
Check and adjust type flag in `AggFieldType`

### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects
- Breaking backward compatibility

### Release note 
- Fix wrongly set NotNullFlag for result type of functions like `case when` expression

### CODE REVIEW

File: types/field_type.go [hunk 1]
```diff
@@ -308,6 +309,15 @@ func MergeFieldType(a byte, b byte) byte {
 	return fieldTypeMergeRules[ia][ib]
 }
 
+// mergeTypeFlag merges two MySQL type flag to a new one
+// currently only NotNullFlag is checked
+// todo more flag need to be checked, for example: UnsignedFlag
+func mergeTypeFlag(a, b uint) uint {
+	ret := a
```
Comment (XuHuaiyu) [inline types/field_type.go:316]: `return a & (b & mysql.NotNullFlag | ^mysql.NotNullFlag)` is ok ?

Comment (windtalker) [reply types/field_type.go]: Yes.

**Pull request merged**